lib/mongo/auth/stringprep.rb

```diff
@@ -0,0 +1,66 @@
+require 'mongo/auth/stringprep/tables'
+
+module Mongo
+  module Auth
+    module StringPrep
+      extend self
+
+      def prepare(data, mappings, prohibited, options = {})
+        out = apply_maps(data, mappings)
+        normalize(out) if options[:normalize]
+        check_prohibited(out, prohibited)
+        check_bidi(out) if options[:bidi]
+        out
+      end
+
+      private
+
+      def apply_maps(data, mappings)
+        out = ''
+
+        data.each_char do |c|
```

Comment:
The more idiomatic Ruby way would be to use inject here.
So ``data.each_char.inject('') do |out, c|``...etc

Reply:
Out of curiosity, is it preferable to use `inject` over `reduce`? `reduce` is more similar to what other languages generally call this operation, but I'm not sure if the Ruby community prefers `inject`